File: Leetcode_solutions/Moving Average from Data Stream.py
class MovingAverage:

    def __init__(self, size: int):
        """
        Initialize your data structure here.
        """
        self.window = []
        self.size = size

    def next(self, val: int) -> float:
        if len(self.window) < self.size:
            self.window.append(val)
            return sum(self.window) / len(self.window)
        else:
            self.window.pop(0)
            self.window.append(val)
            return sum(self.window) / len(self.window)


# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)

File: Leetcode_solutions/Moving Average from Data Stream.java
import java.util.ArrayList;
import java.util.List;

class MovingAverage {
    int size;
    List queue = new ArrayList<Integer>();

    /** Initialize your data structure here. */
    public MovingAverage(int size) {
        this.size = size;
    }

    public double next(int val) {
        queue.add(val);
        int windowSum = 0;
        for (int i = Math.max(0, queue.size() - size); i < queue.size(); ++i) {
            windowSum += (int) queue.get(i);
        }
        return windowSum * 1.0 / Math.min(queue.size(), size);
    }
}
